pkg/transport/generic_transport_controller.go

```diff
@@ -608,6 +610,25 @@ func cleanObject(object *unstructured.Unstructured) *unstructured.Unstructured {
 	delete(annotations, "kubectl.kubernetes.io/last-applied-configuration")
 	objectCopy.SetAnnotations(annotations)
 
-	return objectCopy
+	gvk := objectCopy.GroupVersionKind()
+	if gvk.Group != "apps" || gvk.Kind != "Job" || !strings.HasPrefix(gvk.Version, "v1") {
```

Comment:
Note that we have also another known case where we need to clean an object for a Kubernetes service - see https://github.com/kubestellar/kubestellar/issues/1451